internals-js/src/federation.ts

```diff
@@ -486,6 +496,10 @@ export class FederationMetadata {
     return this.sharingPredicate()(field);
   }
 
+  isFieldKey(field: FieldDefinition<any>): boolean {
```

Comment:
I'd prefer `isKeyField` for this. Because in really this method does not guarantee that the field is a key all by himself, only that it is part of some key, and I'd rather avoid any risk of confusion (I do get that this might be for consistency with `isFieldExternal` but the same problem does not apply for external).